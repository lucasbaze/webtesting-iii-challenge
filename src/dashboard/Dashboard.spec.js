// Test away
import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import { mount, shallow } from 'enzyme';

import Dashboard from './Dashboard';

test('Dashboard renders with controls and display', () => {
    expect(mount(<Dashboard />)).toMatchSnapshot();
    expect(shallow(<Dashboard />)).toMatchSnapshot();
});

test('buttons text changes to reflect the state the door will be in if clicked', async () => {
    const component = render(<Dashboard />);

    let button = component.getByText('Close Gate');
    fireEvent.click(button);

    button = await component.getByText('Open Gate');
    expect(button);
});
