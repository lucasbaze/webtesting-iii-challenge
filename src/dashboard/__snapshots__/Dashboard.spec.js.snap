// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders with controls and display 1`] = `
<Dashboard>
  <Display
    closed={false}
    locked={false}
  >
    <div
      className="display panel"
    >
      <div
        className="led green-led"
      >
        Unlocked
      </div>
      <div
        className="led green-led"
      >
        Open
      </div>
    </div>
  </Display>
  <Controls
    closed={false}
    locked={false}
    toggleClosed={[Function]}
    toggleLocked={[Function]}
  >
    <div
      className="controls panel"
    >
      <button
        className="toggle-btn"
        disabled={true}
        onClick={[Function]}
      >
        Lock Gate
      </button>
      <button
        className="toggle-btn"
        disabled={false}
        onClick={[Function]}
      >
        Close Gate
      </button>
    </div>
  </Controls>
</Dashboard>
`;

exports[`Dashboard renders with controls and display 2`] = `
<Fragment>
  <Display
    closed={false}
    locked={false}
  />
  <Controls
    closed={false}
    locked={false}
    toggleClosed={[Function]}
    toggleLocked={[Function]}
  />
</Fragment>
`;
